src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/AllocationService.kt

```diff
@@ -72,11 +72,12 @@ class AllocationService(
       ?: newAllocation(request, offenderBooking, courseActivity, requestedPayBand)
 
     return allocation.apply {
+      startDate = if (allocation.startDate > LocalDate.now()) request.startDate else allocation.startDate
```

Comment:
Only allow the start date of an allocation to be updated if it hasn't started yet.